### ISSUE TITLE
Empty frontmatter adds extra header

### ISSUE THREAD

**Issue by tuncbkose** [quarto-dev/quarto-cli#10519]
### Bug description

In a book project, adding an empty frontmatter to a document while using a level-one header (`#`) for titles results in an additional level-one header being rendered (both in html and pdf).

### Steps to reproduce

_quarto.yml
```yml
project:
  title: "bug"
  type: "book"

book:
  chapters:
    - index.qmd
    - bug.qmd

format:
  html: default
  pdf: default
```

index.qmd
```qmd
# Page 1
Quarto enables you to weave together content and executable code into a finished document.
```

bug.qmd
```qmd
---
---

# Page 2
Quarto enables you to weave together content and executable code into a finished document.
```

### Expected behavior

No unintended headers should appear.

### Actual behavior

An unintended header appears

### Your environment

- OS: Ubuntu 22.04.4 LTS

### Quarto check output

Quarto 1.4.557
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.12: OK
      Dart Sass version 1.58.3: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.557
      Path: /.../micromamba/envs/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /usr/bin
      Version: 2021

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.5 (Conda)
      Path: /.../micromamba/envs/quarto/bin/python
      Jupyter: 5.7.2
      Kernels: python3, julia-1.10

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/

**Comment by mcanouil** [comment]
Thanks for the report!
Could you try using the latest stable release?

**Comment by tuncbkose** [comment]
Oh oops, didn't notice that, just pulled the latest version in conda-forge I believe. I'll try and update in a bit.

**Comment by mcanouil** [comment]
There are no bugs here in my opinion.

Underlying markdown with `---` means it's a title.
A `---` with empty lines above and below is a horizontal line.

This is markdown syntax, so I don't see how an empty YAML frontmatter would be possible.
Additionally, what is the point to have useless lines at the top of the document?

**Comment by tuncbkose** [comment]
Okay, that's a fair explanation, especially if it is interpreted as markdown instead of an "empty front matter".

I ran into this when I had created a number of files containing the "empty front matter", so that I could modify it as needed. So when I was ready to render the project and some files didn't need any front matter configuration, this happened.  

Expecting` the user to delete it is fair imo. I just wanted to seek clarification, since going from
```md
---
---
```
to
```md
---
key: value
---
```
when the key is not `title` results in expected behavior of headers.
